File: backend/src/main/java/com/coderhan/lastmission/payment/domain/Settlement.java
package com.coderhan.lastmission.payment.domain;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import lombok.Builder;

@Builder
public record Settlement(
        long id,
        long eventId,
        BigDecimal totalSales,
        BigDecimal commissionRate,
        BigDecimal commissionAmount,
        BigDecimal netAmount,
        SettlementStatus status,
        OffsetDateTime settledAt,
        OffsetDateTime createdAt,
        OffsetDateTime updatedAt
) {
}

File: frontend/features/payment/types/settlement.ts
export type SettlementStatus = "COMPLETED";

export interface Settlement {
  id: string;
  eventId: string;
  totalSales: number;
  commissionRate: number;
  commissionAmount: number;
  netAmount: number;
  status: SettlementStatus;
  settledAt: string;
  createdAt: string;
}
